l10n_it_central_journal/wizard/print_giornale.py

```diff
@@ -0,0 +1,168 @@
+        wizard = self
+        if wizard.target_move == "all":
+            target_type = ["posted", "draft"]
+        else:
+            target_type = [wizard.target_move]
+        sql = """
+            SELECT aml.id FROM account_move_line aml
+            LEFT JOIN account_move am ON (am.id = aml.move_id)
+            WHERE
+            aml.date >= %(date_from)s
+            AND aml.date <= %(date_to)s
+            AND am.state in %(target_type)s
+            AND aml.journal_id in %(journal_ids)s
+            ORDER BY am.date, am.name
+        """
+        params = {
+            "date_from": wizard.date_move_line_from,
+            "date_to": wizard.date_move_line_to,
+            "target_type": tuple(target_type),
+            "journal_ids": tuple(self.journal_ids.ids),
+        }
+        self.env.cr.execute(sql, params)
+        res = self.env.cr.fetchall()
+        move_line_ids = flatten(res)
+        return move_line_ids
+
+    def _prepare_datas_form(self):
+        wizard = self
+        datas_form = {}
+        datas_form["date_move_line_from"] = wizard.date_move_line_from
+        datas_form["last_def_date_print"] = wizard.last_def_date_print
+        datas_form["date_move_line_to"] = wizard.date_move_line_to
+        datas_form["fiscal_page_base"] = wizard.fiscal_page_base
+        datas_form["progressive_debit"] = wizard.progressive_debit2
+        datas_form["progressive_credit"] = wizard.progressive_credit
+        datas_form["start_row"] = wizard.start_row
+        datas_form["daterange"] = wizard.daterange.id
+        return datas_form
+
+    def print_giornale(self):
+        wizard = self
+        move_line_ids = self.get_line_ids()
+        if not move_line_ids:
+            raise UserError(_("No documents found in the current selection"))
+        datas_form = self._prepare_datas_form()
+        datas_form["print_state"] = "draft"
+        datas_form["year_footer"] = wizard.year_footer
+        datas = {"ids": move_line_ids, "model": "account.move", "form": datas_form}
+        return self.env.ref(
+            "l10n_it_central_journal.action_report_giornale"
+        ).report_action(self, data=datas)
+
+    def print_giornale_final(self):
+        wizard = self
+        res_company_obj = self.env["res.company"]
+        if (
+            wizard.last_def_date_print
+            and wizard.date_move_line_from <= wizard.last_def_date_print
+        ):
+            raise UserError(_("Date already printed"))
+        else:
+            move_line_ids = self.get_line_ids()
+            if not move_line_ids:
+                raise UserError(_("No documents found in the current selection"))
+            datas_form = self._prepare_datas_form()
+            datas_form["print_state"] = "def"
+            datas_form["year_footer"] = wizard.year_footer
+            datas = {"ids": move_line_ids, "model": "account.move", "form": datas_form}
+            company = res_company_obj.search([("id", "=", self.company_id.id)])
```

Comment:
non la si può recuperare dall'`env`?

Reply:
in verità la company è già quella in self. modifico